---
title: "D3GB package"
author: "Carlos Prieto and David Barrios (University of Salamanca)"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{D3GB package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
## <a id="def">D3GB</a>
D3 Genome Browser is an interactive genome browser that can be easily integrated in R analysis protocols and shared on the Web. It is compatible with popular data formats such as GenBank, GFF, BED, FASTA, and VCF, and enables the exploration of genomic data with a Web browser.
This vignette provides a three examples showing how to create D3GB genome browsers from different file formats. 

## <a id="def">Genome browser from GenBank File</a>
This tutorial shows how to create a D3GB genome browser from a GenBank file with the R package. This will create a complete genome browser of Micromonospora lupini. Install the D3GB package and write the following code in R:


```{r echo=TRUE, message=FALSE, warning=FALSE}
library(D3GB)

# Download GenBank file
gbff <- tempfile()
download.file("ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.gbff.gz",gbff)

# Genome browser generation.
# It creates a genome browser ready to be viewed in Firefox.
# For a server version, ready to be shared with Apache as a Website, set the parameter server=True
gb <- gbk2genomebrowser(gbff, dir = "Micromonospora_Lupini_gbk")
```

